docs/networking/deep-dive.md

```diff
@@ -0,0 +1,112 @@
+---
+sidebar_position: 3
+sidebar_label: Deep Dive
+title: ""
+keywords:
+- Harvester
+- Networking
+- Topology
+---
+# Harvester Network Deep Dive
+
+The network topology below reveals how we implement the Harvester network.
+
+![](/img/v1.1/networking/topology.png)
+
+The diagram contains [the built-in cluster network mgmt](./clusternetwork.md/#built-in-cluster-network) and a [custom cluster network](./clusternetwork.md/#custom-cluster-network) called `oob`.
+
+As shown above, the Harvester network primarily focuses on layer 2. We leverage Linux network devices to construct traffic paths for the communication between VM to VM, VM to host, and VM to external network devices.
+
+The Harvester network is divided into three layers, including:
+
+- KubeVirt networking layer
+
+- Harvester networking layer
+
+- external networking layer
+
+## Kubevirt Networking
+
+The basic idea of KubeVirt is to put libvirt into the Kubernetes pod. The KubeVirt network is responsible for forwarding traffic between the pod and VM. 
+Please refer to the [KubeVirt official document](https://kubevirt.io/2018/KubeVirt-Network-Deep-Dive.html) for more details.
+
+## Harvester Networking
+
+Harvester networking is designed to forward the traffic from pods to the host network. It implements a management network, VLAN networks and untagged networks. We can refer to the last two networks as **bridge networks**, because bridge plays a vital role in their implementation.
+
+### Bridge Network
+
+We leverage [multus CNI](https://github.com/k8snetworkplumbingwg/multus-cni) and [bridge CNI](https://www.cni.dev/plugins/current/main/bridge/) to implement the bridge network.
+
+1. Multus CNI is a Container Network Interface (CNI) plugin for Kubernetes that can attach multiple network interfaces to a pod. Its capability allows a VM to have one NIC for the management network and multiple NICs for the bridge network.
+
+2. Using the bridge CNI, the VM pod will be plugged into the L2 bridge specified in the Network Attachment Definition config.
+
+    ```json
+    # Example 1
+    {
+        "cniVersion": "0.3.1",
+        "name": "vlan100",
+        "type": "bridge",
+        "bridge": "mgmt-br",
+        "promiscMode": true,
+        "vlan": 100,
+    }
+    ```
+
+    ```json
+    # Example 2
+    {
+        "cniVersion": "0.3.1",
+        "name": "untagged-network",
+        "type": "bridge",
+        "bridge": "oob-br",
+        "promiscMode": true,
+        "ipam": {}
+    }
+    ```
+
+    Example 1 is a typical VLAN configuration with VLAN ID 100, while Example 2 is an untagged network configuration with no VLAN ID. The VM pod configured using Example 1 will be plugged into the bridge `mgmt-br`, while the VM pod using Example 2 will be plugged into the bridge `oob-br`.
+
+3. For fault tolerance and high availability concerns, the uplink of each bridge is a bond device. For a bridge port, the uplink should allow the tagged package to pass directly.
```

Comment:
`For fault tolerance and high availability concerns, the uplink of each bridge is a bond device` -> `To achieve high availability and fault tolerance, a bond device is created to serve as the uplink of the bridge, all of the real uplink NICs are binded into this bond device.`

`For a bridge port, the uplink should allow the tagged package to pass directly.` -> `By default, this bond device will allow the target tagged traffic/packet to pass through`.